### PR TITLE
New PR for proofpoint TRAP enhancement

### PR DESCRIPTION
**Pull request by reutshal**
## Status
- [] In Progress
- [X] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/17385
fixes: https://github.com/demisto/etc/issues/14821

## Description
- Improved the description of the *expiration* argument
- The following were removed as integration parameters and are now command arguments.
    - *blacklist_ip*
    - *blacklist_url*
    - *blacklist_domain*
    - *blacklist_hash* 
- Added 4 commands.
    - ***proofpoint-tr-list-incidents***
    - ***proofpoint-tr-get-incident***
    - ***proofpoint-tr-add-comment-to-incident***
    - ***proofpoint-tr-add-user-to-incident***
    - ***proofpoint-tr-ingest-alert***

## Minimum version of Demisto
- [ ] 4.5.0
- [ ] 5.0.0
- [ ] 5.5.0

## Does it break backward compatibility?
   - [X] Yes
       - Changed the integration's parameters
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation 
- [X] Code Review

## Demisto Partner?
- [ ] The title must be in the following format: **[YOUR_PARTNER_ID] short description**

### CODE REVIEW

**Comment by reutshal** [comment]
@ronykoz the only thing that wasn't reviewed is the code from lines 621-668 - 
Added a new command - ingest-alert.
Still need to finish the UT.

**Comment by reutshal** [comment]
@roysagi Can you please go over the files:
- https://github.com/demisto/content/blob/MS_graph-enhancement_new_branch/Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml - All docs approved except the ingest-alert-command.
- https://github.com/demisto/content/blob/MS_graph-enhancement_new_branch/Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse_description.md
-https://github.com/demisto/content/blob/MS_graph-enhancement_new_branch/Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse_CHANGELOG.md
Thanks!

**Comment by roysagi** [comment]
@reutshal @ronykoz - A suggestion: there is the cns_hosts and forensics_hosts in the ingest-alert command which receive a JSON with a list of JSONs in it. 

I believe it would be easier to the user to simply enter the inner CSV of JSON rather then the full JSON.

Instead of:  "{"forensics_hosts": [{"host" : "-", "port": "-"},{"host" : "-", "port": "-"}...]}"
Just this part: {"host" : "-", "port": "-"},{"host" : "-", "port": "-"} - and add the forensics_hosts manually in the code.

It will also make the argument description much clearer...

What do you think?

File: Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml [hunk 1]
```diff
@@ -1,134 +1,892 @@
...
+      - info
+      - ' minor'
+      - ' moderate'
+      - ' major'
+      - ' critical'
+      - ' Informational'
+      - ' Low'
+      - ' Medium'
+      - ' High'
+      - ' Critical'
```
Comment (ronykoz) [inline Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml]: why do we have spaces here?

File: Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml [hunk 2]
```diff
@@ -1,134 +1,892 @@
...
+      - ' attacker_hostname'
+      - ' attacker_machine_name'
+      - ' attacker_user'
+      - ' attacker_mac_address'
+      - ' email_recipient'
+      - ' email_sender'
+      - ' email_subject'
+      - ' message_id'
+      - ' threat_filename'
+      - ' threat_filehash'
```
Comment (ronykoz) [inline Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml]: and here

File: Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml [hunk 3]
```diff
@@ -1,134 +1,892 @@
...
+    execution: false
+    name: proofpoint-tr-add-user-to-incident
+  - arguments:
+    - default: false
+      description: 'POST URL of the JSON alert source. You can find it by navigating
+        to Sources -> JSON event source -> POST URL '
+      isArray: false
+      name: json_source_id
+      required: true
+      secret: false
```
Comment (ronykoz) [inline Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml]: They will have to go to the UI to get that? we should have a solution for this in demisto, they should not go to the product to use it

File: Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml (hunk 1 above)
Comment (reutshal) [reply Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml]: Fixed

File: Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml (hunk 2 above)
Comment (reutshal) [reply Beta_Integrations/ProofpointThreatResponse/ProofpointThreatResponse.yml]: Fixed

**Pull request merged**